pybamm/expression_tree/concatenations.py

```diff
@@ -133,72 +131,32 @@ def create_slices(self, node, mesh):
             start = end
         return slices
 
-    def process_node_for_concatenate(self, node, mesh):
-        """
-        the node is assumed to be constant in time. this function replaces it with a
-        single Vector node with the correct length vector (according to its domain)
-
-        Parameters
-        ----------
-        node: derived from :class:`Symbol`
-            the sub-expression to process (node.is_constant() is true)
-
-        """
-        try:
-            node_size = node.size
-        except AttributeError:
-            node_size = 0
-
-        if node_size > 1:
-            if node_size == sum([mesh[dom].npts for dom in node.domain]):
-                return node
-            else:
-                raise ValueError(
-                    "Error: expression evaluated to a vector of incorrect length"
-                )
-        else:
-            warnings.warn("Implementation not robust")
-            npts = {dom: submesh.npts for dom, submesh in mesh.items()}
-            return pybamm.NumpyBroadcast(node, node.domain, npts)
-
     def evaluate(self, t=None, y=None):
         """ See :meth:`pybamm.Symbol.evaluate()`. """
-
+        # return np.concatenate([child.evaluate(t, y) for child in self.children])
         # preallocate vector
-        vector = np.empty(self._size)
+        if t is None or isinstance(t, numbers.Number):
```

Comment:
Ah ok, you're right, there is actually never any need to work with `y` as a 2D matrix. This has been a headache to implement and would probably be a headache to maintain so very happy to get rid of it! 
I was worried that using a for loop might be slow, but now realise that this will never need to happen during the time stepping (by definition) so shouldn't be an issue